examples/external-data/tests/customerService.test.ts

```diff
@@ -109,39 +162,176 @@ describe("mock-customer-service", () => {
 			}
 
 			// Update external data
-			const dataUpdateResponse = await fetch(
-				`http://localhost:${externalDataServicePort}/set-tasks/${externalTaskListId}`,
+			const taskDataUpdate = {
+				42: {
+					name: "Determine the meaning of life",
+					priority: 37,
+				},
+			};
+			const dataUpdateResponse = await updateExternalData(taskDataUpdate, externalTaskListId);
+			if (dataUpdateResponse.status !== 200) {
+				fail(`Data update failed. Code: ${dataUpdateResponse.status}`);
+			}
+
+			// Delay for a bit to ensure time enough for our webhook listener to have been called.
+			await delay(1000);
+
+			// Verify our listener was notified of data change.
+			expect(wasFluidNotifiedForChange).toBe(true);
+			expect(webhookChangeNotification).toMatchObject({
+				data: taskDataUpdate,
+			});
+		} catch (error) {
+			fail(error);
+		} finally {
+			await closeServer(localService);
+		}
+	});
+	/* eslint-enable @typescript-eslint/no-non-null-assertion */
+
+	it("register-session-url: Complete data flow", async () => {
+		// Set up mock local service, which will be registered as webhook listener
+		const localServiceApp = express();
```

Comment:
Nit: It might be nice to add a helper function for initializing the local service, which could capture all 3 lines here. This same pattern is duplicated quite a bit. (I know this pattern pre-dated this PR, so feel free to close with no action).

Reply:
Done.